### PR TITLE
Feature Updates

### PR DESCRIPTION
**Pull request by blakeneyops**
Added some new capabilities to the launch configuration and auto scaling group parameters.

### CODE REVIEW

File: examples/basic/main.tf [hunk 1]
```diff
@@ -110,7 +123,9 @@ resource "template_file" "user_data" {
 
 ## Provisions basic autoscaling group
 module "example" {
-  source = "github.com/unifio/terraform-aws-asg//group/basic"
+  # Example GitHub source
+  #source = "github.com/unifio/terraform-aws-asg//group/basic"
+  source = "../../group/basic"
```
Comment (yuhunglin) [inline examples/basic/main.tf]: why the change?

File: examples/standard/main.tf [hunk 2]
```diff
@@ -146,6 +146,19 @@ resource "aws_elb" "elb" {
 }
 
 ## Adds security group rules
+resource "aws_security_group_rule" "egress" {
```
Comment (yuhunglin) [inline examples/standard/main.tf]: 👍  good reminder.

File: group/basic/main.tf [hunk 3]
```diff
@@ -1,7 +1,8 @@
 # AWS Autoscaling group
 
 ## Creates security group
-resource "aws_security_group" "sg_asg" {
+resource "aws_security_group" "asg-sg" {
+  name_prefix = "${var.stack_item_label}-"
```
Comment (yuhunglin) [inline group/basic/main.tf]: maybe `${var.stack_item_label}-asg-sg` to match the `stack_item_label` usage we have elsewhere, making it less likely we'll forget to append the naming convention?

File: group/basic/main.tf [hunk 4]
```diff
@@ -11,29 +12,24 @@ resource "aws_security_group" "sg_asg" {
...
-  name                 = "${var.stack_item_label}-lc"
-  image_id             = "${var.ami}"
-  instance_type        = "${var.instance_type}"
-  iam_instance_profile = "${var.instance_profile}"
-  key_name             = "${var.key_name}"
-  security_groups      = ["${aws_security_group.sg_asg.id}"]
-  enable_monitoring    = "${var.enable_monitoring}"
-  ebs_optimized        = "${var.ebs_optimized}"
-  user_data            = "${var.user_data}"
+  name_prefix                 = "${var.stack_item_label}-"
```
Comment (yuhunglin) [inline group/basic/main.tf]: same with `${var.stack_item_label}-lc-`

File: group/basic/main.tf [hunk 5]
```diff
@@ -1,7 +1,8 @@
 # AWS Autoscaling group
 
 ## Creates security group
-resource "aws_security_group" "sg_asg" {
+resource "aws_security_group" "asg-sg" {
```
Comment (yuhunglin) [inline group/basic/main.tf]: nitpick: what's our stance on variable naming in hcl?

File: group/standard/main.tf [hunk 6]
```diff
@@ -1,7 +1,8 @@
 # AWS Autoscaling group
 
 ## Creates security group
-resource "aws_security_group" "sg_asg" {
+resource "aws_security_group" "asg-sg" {
```
Comment (yuhunglin) [inline group/standard/main.tf]: more 🐍  vs 💨  case fun.

File: group/standard/main.tf [hunk 7]
```diff
@@ -11,29 +12,24 @@ resource "aws_security_group" "sg_asg" {
...
-  name                 = "${var.stack_item_label}-lc"
-  image_id             = "${var.ami}"
-  instance_type        = "${var.instance_type}"
-  iam_instance_profile = "${var.instance_profile}"
-  key_name             = "${var.key_name}"
-  security_groups      = ["${aws_security_group.sg_asg.id}"]
-  enable_monitoring    = "${var.enable_monitoring}"
-  ebs_optimized        = "${var.ebs_optimized}"
-  user_data            = "${var.user_data}"
+  name_prefix                 = "${var.stack_item_label}-"
```
Comment (yuhunglin) [inline group/standard/main.tf]: `${var.stack_item_label}-lc-`

File: group/standard/variables.tf [hunk 8]
```diff
@@ -41,6 +45,10 @@ variable "ebs_optimized" {
   default = false
 }
 
+variable "placement_tenancy" {
+  default = "default"
```
Comment (yuhunglin) [inline group/standard/variables.tf]: Hmm, had a fun thought about limiting inputs. What if we converted some of our variables with a fixed set of inputs to maps that would do an identity map? It would probably push errors up from runtime to compile time.

**Comment by yuhunglin** [comment]
LGTM, lots of minor things, mainly around the var naming convention.

File: examples/basic/main.tf (hunk 1 above)
Comment (blakeneyops) [inline examples/basic/main.tf]: Doesn't actually test what's in the repo until it's committed, which is a pain when you're developing and want to use these to verify.

File: group/basic/main.tf (hunk 3 above)
Comment (blakeneyops) [inline group/basic/main.tf]: I wanted to see what this actually results in. I've not tried it just yet, but generally agree with your recommendation.

**Comment by blakeneyops** [comment]
Regarding name convention, I checked around and it looks like underscores are more the norm. I pretty much follow a <resource_type>_<context> pattern. If there is no specific context, I go with just that. Regarding actual naming and tagging in AWS, I've removed resource type label from resources where it is implied. For example, I don't need to have `lc` in the name of a launch configuration, etc. Have a look at the updates here and see what you think and let me know if you have any questions.

File: examples/basic/main.tf (hunk 1 above)
Comment (yuhunglin) [inline examples/basic/main.tf]: 👍  Maybe can sub out the prefix with a variable that can be ovewridden locally?

**Comment by yuhunglin** [comment]
Sounds good, that sounds like the general consensus naming convention for terraform.

For naming and tagging: 🆗,, that sounds reasonable. At the same time, avoiding building up exceptions to the rule unless it provides some sizable benefit makes it easier to enforce conventions.

LGTM, feel free to toy with the map idea if you'd like.
